### ISSUE TITLE
Cleanup mpl circuit_drawer module (post 0.7)

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-terra#1452]
### What is the expected enhancement?

As pointed out in the review for https://github.com/Qiskit/qiskit-terra/pull/1450 there are a few artifacts leftover from the json->dag conversion for the mpl backend. Things like misleading variable names and out of date notes and comments. After the 0.7 release it would be good to go through the module and clean this up to make maintenance easier.

**Comment by t-imamichi** [comment]
@nkanazawa1989 @mtreinish OK. Let's discuss about it together.

**Comment by mtreinish** [comment]
@t-imamichi @nkanazawa1989 sorry I missed your comment here before. I just pushed #1651 which does the cleanups that I opened this issue about. (which was just an out of date comment and an inaccurate field name) There might be other things in there, I didn't do a thorough scan of the code to check. If you had other ideas on what could/should be cleaned up and/or updated in the matplotlib drawer since the 0.7 release we can continue to discuss them here (or we can open a new issue for that).
